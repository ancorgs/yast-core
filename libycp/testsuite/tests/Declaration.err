Parsed:
----------------------------------------------------------------------
{
  global define hallo(integer x, [boolean bl, map mp, byteblock b] st) ``{
  y2milestone ("%1 %2", x, st);
  return select (st, 1, nil);
}
;
  return (hallo (17, [true, $[1:45], #[102A]]) == $[1:45]);
}
----------------------------------------------------------------------
[YCP] tests/Declaration.ycp(hallo):3 17 [true, $[1:45], #[102A]]
Parsed:
----------------------------------------------------------------------
(hallo (19, [true, $[1:2], #[DEADBEAF]]) == $[1:2])
----------------------------------------------------------------------
[YCP] tests/Declaration.ycp(hallo):3 19 [true, $[1:2], #[DEADBEAF]]
Parsed:
----------------------------------------------------------------------
(hallo (120) == nil)
----------------------------------------------------------------------
[Interpreter] tests/Declaration.ycp:4 actual arguments don't match definition for 'hallo'
  Expected hallo(integer x, [boolean bl, map mp, byteblock b] st)
  Seen hallo (120)
Parsed:
----------------------------------------------------------------------
(hallo (18, [false, 1, $[]]) == nil)
----------------------------------------------------------------------
[Interpreter] tests/Declaration.ycp:4 actual arguments don't match definition for 'hallo'
  Expected hallo(integer x, [boolean bl, map mp, byteblock b] st)
  Seen hallo (18, [false, 1, $[]])
Parsed:
----------------------------------------------------------------------
{
  define withnil(integer a, boolean b) ``{
  return [a, b];
}
;
  return withnil (42, nil);
}
----------------------------------------------------------------------
