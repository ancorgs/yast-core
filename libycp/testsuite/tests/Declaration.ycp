{
    global define hallo(integer x, [ boolean bl, map mp, byteblock b] st) ``{
	y2milestone ("%1 %2", x, st);
	return select(st, 1, nilmap);
    };
    return hallo(17, [ true, $[1:45], #[102a]]) == $[1:45];
}

( hallo(19, [ true, $[1:2], #[DeadBeaf] ]) == $[1:2])
( hallo(120) == nil)
( hallo(18, [ false, 1, $[] ] ) == nil)

{
    define withnil (integer a, boolean b) ``{
	return [a,b];
    }
    // should return [42, nil]
    return withnil (42, nil);
}
