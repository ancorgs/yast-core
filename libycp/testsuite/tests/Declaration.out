(true)
(true)
(true)
(true)
([42, nil])
